src/c++/libos/lwip/lwip_queue.cc

```diff
@@ -1506,7 +1506,7 @@ dmtr::lwip_queue::parse_packet(struct sockaddr_in &src,
 
         char *offset_buf = static_cast<char*>(sga.sga_buf);
         for (size_t i = 0; i < sga.sga_numsegs; ++i) {
-            uint32_t seg_len = ntohl(*(uint32_t*)offset_buf);
+            uint32_t seg_len = ntohl(*reinterpret_cast<uint32_t*>(offset_buf));
```

Comment:
Shouldn't that be a static cast, if reinterpret cast is made for "complex" data structures (e.g. non native types)?